File: 2020/06/1.hs
import           Data.List                      ( nub )
import           Data.List.Split                ( splitOn )

main :: IO ()
main =
  readFile "input"
    >>= print
    .   sum
    .   map (length . nub . filter (/= '\n'))
    .   splitOn "\n\n"

File: 2024/07/1.jl
function anyop(t, ns)
    length(ns) == 1 && return only(ns) == t
    return anyop(t - ns[end], ns[1:end-1]) ||
           (t % ns[end] == 0 && anyop(t ÷ ns[end], ns[1:end-1]))
end

i = map(fs -> (parse(Int, fs[1]), parse.(Int, split(fs[2]))),
        split.(readlines("input"), ':'))
println(sum(fs -> anyop(fs[1], fs[2]) * fs[1], i))
